### PR TITLE
refactor(v-expansion-panel): converted to typescript

### PR DESCRIPTION
**Pull request by nekosaur**
## Description
Converted v-expansion-panel to typescript. Had to disable two tests because of https://github.com/vuejs/vue-test-utils/issues/824

## Motivation and Context
because!

## How Has This Been Tested?

visually, jest

## Markup:

<details>

```vue
<template>
  <v-app>
    <div class="text-xs-center mb-3">{{ panel }}</div>
    <v-expansion-panel
      v-model="panel"
      expand
    >
      <v-expansion-panel-content
        v-for="(item, i) in 5"
        :key="i"
      >
        <div slot="header">Item</div>
        <v-card>
          <v-card-text class="grey lighten-3">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</v-card-text>
        </v-card>
      </v-expansion-panel-content>
    </v-expansion-panel>

    <div class="d-flex justify-between align-center mb-3">
      <v-btn @click="all">all</v-btn>
      <v-btn @click="none">none</v-btn>
    </div>

    <v-expansion-panel
      v-model="panel2"
      expand
    >
      <v-expansion-panel-content
        v-for="(item,i) in items"
        :key="i"
      >
        <div slot="header">Item</div>
        <v-card>
          <v-card-text>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</v-card-text>
        </v-card>
      </v-expansion-panel-content>
    </v-expansion-panel>

    <v-expansion-panel>
      <v-expansion-panel-content
        v-for="(item,i) in 5"
        :key="i"
        expand-icon="mdi-menu-down"
      >
        <div slot="header">Item</div>
        <v-card>
          <v-card-text class="grey lighten-3">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</v-card-text>
        </v-card>
      </v-expansion-panel-content>
    </v-expansion-panel>
  </v-app>
</template>

<script>
  export default {
    data () {
      return {
        panel: [false, true, true],
        panel2: [],
        items: 5
      }
    },
    methods: {
      // Create an array the length of our items
      // with all values as true
      all () {
        this.panel2 = [...Array(this.items).keys()].map(_ => true)
      },
      // Reset the panel
      none () {
        this.panel2 = []
      }
    }
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: src/components/VExpansionPanel/VExpansionPanel.ts [hunk 1]
```diff
@@ -1,15 +1,21 @@
 import '../../stylus/components/_expansion-panel.styl'
 
+import { VExpansionPanelContent } from '.'
+
 import Themeable from '../../mixins/themeable'
 import { provide as RegistrableProvide } from '../../mixins/registrable'
 
+import mixins from '../../util/mixins'
+import { VNode } from '../../../node_modules/vue'
+import { PropValidator } from '../../../node_modules/vue/types/options'
```
Comment (KaelWD) [inline src/components/VExpansionPanel/VExpansionPanel.ts]: Use absolute paths

File: src/components/VExpansionPanel/VExpansionPanel.ts [hunk 2]
```diff
@@ -25,16 +31,16 @@ export default {
     value: {
       type: [Number, Array],
       default: () => null
-    }
+    } as any as PropValidator<number | number[]>
```
Comment (KaelWD) [inline src/components/VExpansionPanel/VExpansionPanel.ts]: Is `any` needed here?

File: src/components/VExpansionPanel/VExpansionPanel.ts [hunk 3]
```diff
@@ -46,10 +52,10 @@ export default {
 
   watch: {
     expand (val) {
```
Comment (KaelWD) [inline src/components/VExpansionPanel/VExpansionPanel.ts]: annotate watch params

File: src/components/VExpansionPanel/VExpansionPanel.ts (hunk 2 above)
Comment (KaelWD) [reply src/components/VExpansionPanel/VExpansionPanel.ts]: ok that's fine then, just checking.

File: src/components/VExpansionPanel/VExpansionPanelContent.ts [hunk 4]
```diff
@@ -3,19 +3,40 @@ import { VExpandTransition } from '../transitions'
...
 import Toggleable from '../../mixins/toggleable'
 import Rippleable from '../../mixins/rippleable'
-import { inject as RegistrableInject } from '../../mixins/registrable'
+import { Registrable, inject as RegistrableInject } from '../../mixins/registrable'
 
 import VIcon from '../VIcon'
+import VExpansionPanel from '.'
+
+import mixins, { ExtractVue } from '../../util/mixins'
+import Vue, { VNode } from '../../../node_modules/vue'
```
Comment (KaelWD) [inline src/components/VExpansionPanel/VExpansionPanelContent.ts]: Another one

Comment (nekosaur) [reply src/components/VExpansionPanel/VExpansionPanelContent.ts]: dammit. seems like it's the vscode 1.25 update that's causing it

**Pull request merged**